metron-platform/metron-parsers/src/main/java/org/apache/metron/parsers/interfaces/MessageParser.java

```diff
@@ -34,15 +34,15 @@
    * @param rawMessage
    * @return If null is returned, this is treated as an empty list.
    */
-  List<T> parse(byte[] rawMessage);
+  List<T> parse(byte[] rawMessage, SensorParserConfig sensorParserConfig);
```

Comment:
On second thought, maybe instead of changing `parseOptionally`, you should just call `configurationChanged` from the bolt itself upon configuration change.